RocketChat/Rocket.Chat.ReactNative
# [NEW] Go to RoomView if the mentioned #channel is already joined

## Proposed changes


## Issue(s)	

Fixes #4201 
## How to test or reproduce

* Mention a `#channel` that you haven't joined and click on it. You should be redirected to `RoomInfoView`  .
* Mention a `#channel` that you have already joined and click on it. You should be redirected to the `RoomView` directly.


## Screenshots

https://user-images.githubusercontent.com/77742477/169062333-ab52836a-797e-4bc4-b537-9e59de789923.mp4

## Types of changes




## Checklist



## Further comments
